src/js/components/SearchBar.jsx

```diff
@@ -49,57 +51,130 @@ const _sameSearchTerm = (value, old) => value.trim() === old.trim();
 export class SearchBar extends PureComponent {
   constructor(props) {
     super(props);
-    this.state = { value: '' };
+    this.state = { value: '', startsWithSearch: false };
   }
 
   componentDidMount() {
     // on start focus the input to hint for a user to search
     this.textField.focus();
   }
 
-  handleSearch(e, value) {
+  componentDidUpdate(previousProps, previousState) {
+    // re-run search?
+    if (previousState.startsWithSearch !== this.state.startsWithSearch) {
+      this.handleSearch(null, this.state.value, true);
+    }
+  }
+
+  buildApiUrl(state = this.state) {
+    const baseSearchUrl = `${SEARCH_URL}${state.value.toLowerCase()}`;
```

Comment:
from the docblock of handleSearch describing 'value', searchTerm or currentSearch would be a better name for it in state. Assuming it's talking about the same thing.